core/arch/arm/kernel/spmc_sp_handler.c

```diff
@@ -579,6 +579,72 @@ static void ffa_mem_retrieve(struct thread_smc_args *args,
 	ffa_set_error(args, ret);
 }
 
+static void ffa_mem_relinquish(struct thread_smc_args *args,
+			       struct sp_session *caller_sp,
+			       struct ffa_rxtx  *rxtx)
+{
+	struct sp_mem *smem = NULL;
+	struct ffa_mem_relinquish *mem = rxtx->rx;
+	struct sp_mem_receiver *receiver = NULL;
+	TEE_Result err = FFA_OK;
+	uint32_t exceptions = 0;
+
+	if (!check_rxtx(rxtx)) {
+		ffa_set_error(args, FFA_DENIED);
+		return;
+	}
+
+	cpu_spin_lock(&rxtx->spinlock);
```

Comment:
Unless you know that interrupts are masked you should use `cpu_spin_lock_xsave()` instead.

Reply:
Done